pennylane/qchem/convert.py

```diff
@@ -1007,3 +1007,110 @@ def _uccsd_state(ccsd_solver, tol=1e-15):
+    contribution from the doubly-excited state, i.e., :math:`|0 0 1 1 \rangle`, will be
+    ``{(1, 1): 0.99, (2, 2): 0.01}``.
+
+    The determinants and coefficients should be supplied externally. They should be calculated by
+    using Block2 DMRGDriver's `get_csf_coefficients()` method.
+
+    Args:
+        wavefunction tuple(array[int], array[float]): determinants and coefficients in physicist notation
+        tol (float): the tolerance for discarding Slater determinants with small coefficients
+
+    Returns:
+        dict: dictionary of the form `{(int_a, int_b) :coeff}`, with integers `int_a, int_b`
+        having binary representation corresponding to the Fock occupation vector in alpha and beta
+        spin sectors, respectively, and coeff being the CI coefficients of those configurations
+
+    **Example**
+
+    >>> from pyscf import gto, scf
+    >>> from pyblock2 import DMRGDriver
+    >>> mol = gto.M(atom=[['Li', (0, 0, 0)], ['Li', (0,0,0.71)]], basis='sto6g', symmetry="d2h")
+    >>> myhf = scf.RHF(mol).run()
+    >>> ncas, n_elec, spin, ecore, h1e, g2e, orb_sym = itg.get_uhf_integrals(mf, ncore, ncas, g2e_symm=8)
+    >>> driver = DMRGDriver(scratch=dir, symm_type=SymmetryTypes.SZ, n_threads=n_threads, stack_mem=mem)
+    >>> driver.initialize_system(n_sites=ncas, n_elec=n_elec, spin=spin, orb_sym=orb_sym)
+    >>> mpo = driver.get_qc_mpo(h1e=h1e, g2e=g2e, ecore=ecore+eshift, reorder=reorder, iprint=iprint)
+    >>> ket = driver.get_random_mps(tag="GS", bond_dim=schedule[0][0], occs=occs, nroots=nroots, dot=dot)
+    >>> energies = driver.dmrg(mpo, ket, n_sweeps=n_sweeps[ii], bond_dims=Mvals, noises=noisevals, thrds=thrdsvals, iprint=iprint, tol=tol)
+    >>> wavefunction = driver.get_csf_coefficients(ket)
+    >>> wf_dmrg = _dmrg_state(wavefunction, tol=1e-1)
+    """
+    dets, coeffs = wavefunction
+
+    row, col, dat = [], [], []
+
+    for ii, det in enumerate(dets):
+        stra, strb = _sitevec_to_fock(det)
+        row.append(stra)
+        col.append(strb)
+
+        # compute and fix parity to stick to pyscf notation
+        lsta = np.array(list(map(int, bin(stra)[2:])))[::-1]
+        lstb = np.array(list(map(int, bin(strb)[2:])))[::-1]
+
+        # pad the relevant list
+        maxlen = max([len(lsta), len(lstb)])
+        lsta = np.pad(lsta, (0, maxlen - len(lsta)))
+        lstb = np.pad(lstb, (0, maxlen - len(lstb)))
+
+        which_occ = np.where(lsta == 1)[0]
+        if len(which_occ) == 0:
+            parity = 1.0
+        else:
+            parity = (-1) ** np.sum([np.sum(lstb[: int(ind)]) for ind in which_occ])
+        dat.append(parity * coeffs[ii])
+
+    ## create the FCI matrix as a dict
+    dict_fcimatr = dict(zip(list(zip(row, col)), dat))
+
+    # filter based on tolerance cutoff
+    dict_fcimatr = {key: value for key, value in dict_fcimatr.items() if abs(value) > tol}
+
+    return dict_fcimatr
```

Comment:
So is there no explicit dependency on block2?

Reply:
No. We do not have any external dependency for importing initial states.

Reply:
I'm a little confused for this part. So here the `wavefunction` object is treated like a `block2` object, but we don't need access to `block2` to inpect the object?

How can we get away with not needing `block2` in the tests?

Reply:
It is not a `block2` object. The nice thing about block2 is the `driver.get_csf_coefficients()` method which gives the wave function as a tuple of two arrays (coefficients and slater determinants). The function we are adding here, converts that format to a state vector. I will clarify that in the example.

Reply:
Thanks, that makes sense! How hard is `block2` to install? I'd be tempted to have it installed and used it the tests, so we can know in future if the output format is ever changed by `block2`. Not a huge blocker, especially if it's hard to install or we're confident in the stability of the library.

Reply:
Installing `block2` is not very difficult but installing `dice`, needed for SHCI calculations, is extremely difficult. I suggest not installing any library other than `pyscf` which is already a soft dependency for building molecular Hamiltonians.